src/gui/trackersadditiondialog.cpp

```diff
@@ -74,14 +74,35 @@ TrackersAdditionDialog::~TrackersAdditionDialog()
     delete m_ui;
 }
 
-void TrackersAdditionDialog::onAccepted() const
+bool isValidEndpoint(const QStringView endpoint)
+{
+    if (endpoint.isEmpty())
+        return false;
+    const QUrl url {endpoint.toString()};
+    if (!url.isValid())
+        return false;
+    if (url.scheme().isEmpty())
+        return false;
+    if (url.host().isEmpty())
+        return false;
+    return true;
+}
+
+void TrackersAdditionDialog::onAccepted()
 {
     const QList<BitTorrent::TrackerEntryStatus> currentTrackers = m_torrent->trackers();
     const int baseTier = !currentTrackers.isEmpty() ? (currentTrackers.last().tier + 1) : 0;
 
     QList<BitTorrent::TrackerEntry> entries = BitTorrent::parseTrackerEntries(m_ui->textEditTrackersList->toPlainText());
     for (BitTorrent::TrackerEntry &entry : entries)
+    {
+        if (!isValidEndpoint(entry.url))
+        {
+            QMessageBox::warning(this, tr("Invalid tracker URL"), tr("The tracker URL \"%1\" is invalid").arg(entry.url));
+            return;
```

Comment:
IMO, it looks unfriendly if you reject all the entered trackers because of invalid one. Or am I misunderstood it?